src/index.ts

```diff
@@ -274,7 +274,7 @@ const typescript: PluginImpl<RPT2Options> = (options) =>
 			// Rollup can't see these otherwise, because they are "emit-less" and produce no JS
 			if (result.references && supportsThisLoad) {
 				for (const ref of result.references) {
-					if (ref.endsWith(".d.ts"))
```

Comment:
So I believe I _intentionally_ wrote this as hard-coded, to match [this line](https://github.com/ezolenko/rollup-plugin-typescript2/blob/68017aeb9eb81f9473bfe037ece52a837c1986b5/src/index.ts#L210) in `resolveId`. `filter` is still called after that line. 

I believe the intent of that is that rpt2 should _never_ resolve declarations, even if a user overrides the `exclude`.

In the branch I mentioned in https://github.com/ezolenko/rollup-plugin-typescript2/issues/446#issuecomment-1625859100, similar to `resolveId`, I do _both_ a declaration check _and_ then a `filter` check. I then had a [follow-up commit](https://github.com/agilgur5/rollup-plugin-typescript2/commit/15bd6ec3bbf448aac7c02d69c3a360a256724fa3) to refactor those checks into a single function to always be equivalent.

Let me make a PR for that. I'll expand that function to also check all current declaration extensions (i.e. `.d.ts` _as well as_ `.d.cts` and `.d.mts`)

Reply:
Added #456 to fix that